Spigot-Server-Patches/0316-Add-config-to-disable-ender-dragon-legacy-check.patch

```diff
@@ -0,0 +1,41 @@
+From 0551efc4884d57bef5f7460760ec5f94a3882fd5 Mon Sep 17 00:00:00 2001
+From: BillyGalbreath <Blake.Galbreath@GMail.com>
+Date: Fri, 22 Jun 2018 10:38:31 -0500
+Subject: [PATCH] Add config to disable ender dragon legacy check
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+index 99ad40fa..06c54690 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -524,4 +524,9 @@ public class PaperWorldConfig {
+     private void shieldBlockingDelay() {
+         shieldBlockingDelay = getInt("game-mechanics.shield-blocking-delay", 5);
+     }
++
++    public boolean scanForLegacyEnderDragon = true;
++    private void scanForLegacyEnderDragon() {
++        scanForLegacyEnderDragon = getBoolean("game-mechanics.scan-for-legacy-ender-dragon", true);
++    }
+ }
+diff --git a/src/main/java/net/minecraft/server/EnderDragonBattle.java b/src/main/java/net/minecraft/server/EnderDragonBattle.java
+index ce0dcbc6..54c0b70c 100644
+--- a/src/main/java/net/minecraft/server/EnderDragonBattle.java
++++ b/src/main/java/net/minecraft/server/EnderDragonBattle.java
+@@ -42,7 +42,12 @@ public class EnderDragonBattle {
+     public EnderDragonBattle(WorldServer worldserver, NBTTagCompound nbttagcompound) {
+         this.c = (BossBattleServer) (new BossBattleServer(new ChatMessage("entity.EnderDragon.name", new Object[0]), BossBattle.BarColor.PINK, BossBattle.BarStyle.PROGRESS)).setPlayMusic(true).c(true);
+         this.e = Lists.newArrayList();
+-        this.n = true;
++        // Paper start
++        this.n = worldserver.paperConfig.scanForLegacyEnderDragon;
++        if (!this.n) {
++            this.k = true;
```

Comment:
can you use an obf helper here, otherwise we won't know what k is in a major update.